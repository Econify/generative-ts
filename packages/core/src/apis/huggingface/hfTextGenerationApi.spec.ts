import {
  HfTextGenerationTaskApi,
  HfTextGenerationTaskOptions,
} from "./hfTextGenerationApi";

function render(context: Omit<HfTextGenerationTaskOptions, "modelId">) {
  try {
    const rendered = HfTextGenerationTaskApi.requestTemplate.render({
      modelId: "mock-model-id",
      ...context,
    });
    return JSON.parse(rendered);
  } catch (error) {
    throw new Error("Invalid JSON output");
  }
}

describe("HfInferenceApi:", () => {
  test("prompt only", () => {
    const rendered = render({
      prompt: "mock-prompt",
    });

    expect(rendered).toMatchSnapshot();
  });

  test("prompt and parameters", () => {
    const rendered = render({
      prompt: "mock-prompt",
      parameters: {
        top_k: 50,
        top_p: 0.9,
        temperature: 0.7,
        repetition_penalty: 1.2,
        max_new_tokens: 100,
        max_time: 60,
        return_full_text: true,
        num_return_sequences: 3,
        do_sample: true,
      },
    });

    expect(rendered).toMatchSnapshot();
  });

  test("prompt and options", () => {
    const rendered = render({
      prompt: "mock-prompt",
      options: {
        use_cache: true,
        wait_for_model: false,
      },
    });

    expect(rendered).toMatchSnapshot();
  });

  test("prompt, parameters, and options", () => {
    const rendered = render({
      prompt: "mock-prompt",
      parameters: {
        top_k: 50,
        top_p: 0.9,
        temperature: 0.7,
        repetition_penalty: 1.2,
        max_new_tokens: 100,
        max_time: 60,
        return_full_text: true,
        num_return_sequences: 3,
        do_sample: true,
      },
      options: {
        use_cache: true,
        wait_for_model: false,
      },
    });

    expect(rendered).toMatchSnapshot();
  });

  test("empty parameters", () => {
    const rendered = render({
      prompt: "mock-prompt",
      parameters: {},
    });

    expect(rendered).toMatchSnapshot();
  });

  test("empty options", () => {
    const rendered = render({
      prompt: "mock-prompt",
      options: {},
    });

    expect(rendered).toMatchSnapshot();
  });

  test("all options", () => {
    const rendered = render({
      prompt: "mock-prompt",
      parameters: {
        top_k: 50,
        top_p: 0.9,
        temperature: 0.7,
        repetition_penalty: 1.2,
        max_new_tokens: 100,
        max_time: 60,
        return_full_text: true,
        num_return_sequences: 3,
        do_sample: true,
      },
      options: {
        use_cache: true,
        wait_for_model: false,
      },
    });

    expect(rendered).toMatchSnapshot();
  });
});
