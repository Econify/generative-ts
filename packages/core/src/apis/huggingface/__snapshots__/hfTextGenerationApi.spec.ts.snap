// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HfInferenceApi: all options 1`] = `
{
  "inputs": "mock-prompt",
  "options": {
    "use_cache": true,
    "wait_for_model": false,
  },
  "parameters": {
    "do_sample": true,
    "max_new_tokens": 100,
    "max_time": 60,
    "num_return_sequences": 3,
    "repetition_penalty": 1.2,
    "return_full_text": true,
    "temperature": 0.7,
    "top_k": 50,
    "top_p": 0.9,
  },
}
`;

exports[`HfInferenceApi: empty options 1`] = `
{
  "inputs": "mock-prompt",
  "options": {},
}
`;

exports[`HfInferenceApi: empty parameters 1`] = `
{
  "inputs": "mock-prompt",
  "parameters": {},
}
`;

exports[`HfInferenceApi: prompt and options 1`] = `
{
  "inputs": "mock-prompt",
  "options": {
    "use_cache": true,
    "wait_for_model": false,
  },
}
`;

exports[`HfInferenceApi: prompt and parameters 1`] = `
{
  "inputs": "mock-prompt",
  "parameters": {
    "do_sample": true,
    "max_new_tokens": 100,
    "max_time": 60,
    "num_return_sequences": 3,
    "repetition_penalty": 1.2,
    "return_full_text": true,
    "temperature": 0.7,
    "top_k": 50,
    "top_p": 0.9,
  },
}
`;

exports[`HfInferenceApi: prompt only 1`] = `
{
  "inputs": "mock-prompt",
}
`;

exports[`HfInferenceApi: prompt, parameters, and options 1`] = `
{
  "inputs": "mock-prompt",
  "options": {
    "use_cache": true,
    "wait_for_model": false,
  },
  "parameters": {
    "do_sample": true,
    "max_new_tokens": 100,
    "max_time": 60,
    "num_return_sequences": 3,
    "repetition_penalty": 1.2,
    "return_full_text": true,
    "temperature": 0.7,
    "top_k": 50,
    "top_p": 0.9,
  },
}
`;
